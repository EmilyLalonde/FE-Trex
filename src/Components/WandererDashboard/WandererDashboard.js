import React, { Component } from 'react';
import Map from '../Map/Map'
import Weather from '../Weather/Weather'
import { 
  StyleSheet, 
  Text, 
  View, 
  ScrollView,
  TouchableOpacity 
} from 'react-native';

import WandererHeader from '../WandererHeader/WandererHeader';
import WandererFooter from '../WandererFooter/WandererFooter';

export default class WandererDashboard extends Component {

  constructor(props) {
    super(props)
  }

  render() {
  const {navigate} = this.props.navigation;

    return (
  <View style={styles.container}>
    <WandererHeader />
    <ScrollView>
      <View style={styles.topContainer}>
      <Text style={styles.textEmergency}>Emergency Call</Text>
      <Weather />
      </View>
      <Map />
    </ScrollView>
    <WandererFooter navigate={navigate} userId={this.props.navigation.getParam('userId')}/>
  </View>
      
    )
  }
}

const styles = StyleSheet.create({
  text: {
    color: 'white',
    marginVertical: 40,
    textAlign: 'center',
    fontSize: 30,
    width: 'auto'
  },
  container: {
    flex: 1,
    backgroundColor: '#000000',
    alignItems: 'stretch',
    justifyContent: 'flex-start'
  },
  textEmergency: {
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 8,
    borderStyle: 'solid',
    color: 'white',
    height: 200,
    marginVertical: 15,
    textAlign: 'center',
    fontSize: 20,
    width: 170
  },
  topContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
});