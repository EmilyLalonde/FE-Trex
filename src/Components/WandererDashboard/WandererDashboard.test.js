import React from "react";
import "react-native";
import WandererDashboard from "./WandererDashboard";
import { shallow } from "enzyme";

describe("WandererDashboard", () => {
  it("should return true", () => {
    expect(true).toEqual(true);
  });
  it("should match the snapshot when loading", () => {
    let wrapper = shallow(<WandererDashboard navigation={{ navigate: jest.fn() }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
