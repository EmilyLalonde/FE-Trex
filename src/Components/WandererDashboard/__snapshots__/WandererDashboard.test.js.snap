// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WandererDashboard should match the snapshot when loading 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <WandererHeader />
  <ScrollViewMock>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "white",
            "fontSize": 20,
            "height": 200,
            "marginVertical": 15,
            "textAlign": "center",
            "width": 170,
          }
        }
      >
        Emergency Call
      </Text>
      <Weather />
    </View>
    <Map />
  </ScrollViewMock>
  <WandererFooter
    navigate={[MockFunction]}
  />
</View>
`;
